::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
: Copyright (c) 2018 Mirantis Inc., Enea AB and others.
:
: All rights reserved. This program and the accompanying materials
: are made available under the terms of the Apache License, Version 2.0
: which accompanies this distribution, and is available at
: http://www.apache.org/licenses/LICENSE-2.0
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Mon, 26 Feb 2018 06:11:06 +0100
Subject: [PATCH] network.dpdk: ovs-vswitchd alternative inversion

By default, the alternative installed by `openvswitch-switch`
packages shadows (higher prio) `openvswitch-switch-dpdk` version,
e.g. for UCA Pike packages:

$ update-alternatives --query ovs-vswitchd
Alternative: /usr/lib/openvswitch-switch-dpdk/ovs-vswitchd-dpdk
Priority: 50
Alternative: /usr/lib/openvswitch-switch/ovs-vswitchd
Priority: 100

To avoid confusion, when DPDK is enabled and ovs-vswitchd-dpdk
should be used, remove the alternative for classic ovs-vswitchd
(which can easily be re-eneabled later if needed).

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 linux/network/dpdk.sls | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/linux/network/dpdk.sls b/linux/network/dpdk.sls
index 05fe05f..1e169a8 100644
--- a/linux/network/dpdk.sls
+++ b/linux/network/dpdk.sls
@@ -40,6 +40,15 @@ openvswitch_dpdk_pkgs:
     - openvswitch-switch
     - bridge-utils

+openvswitch_dpdk_ovs_alternative:
+  alternatives.remove:
+  - name: ovs-vswitchd
+  - path: /usr/lib/openvswitch-switch/ovs-vswitchd
+  - require:
+    - pkg: openvswitch_dpdk_pkgs
+  - watch_in:
+    - service: service_openvswitch
+
 linux_network_dpdk_ovs_service:
   cmd.run:
   - name: "ovs-vsctl --no-wait set Open_vSwitch . other_config:dpdk-init=true"
